src/python_routing_v02/build_tests.py

```diff
@@ -161,22 +161,31 @@ def build_test_parameters(
 
 
 def parity_check(parity_parameters, nts, dt, results):
-    validation_files = glob.glob(
-        parity_parameters["parity_check_input_folder"]
-        + parity_parameters["parity_check_file_pattern_filter"],
-        recursive=True,
-    )
+        
+    if "parity_check_input_folder" in parity_parameters.keys():
+        # Glob list of CHRTOUT files for validation
+        validation_files = glob.glob(
+            parity_parameters["parity_check_input_folder"]
+            + parity_parameters["parity_check_file_pattern_filter"],
+            recursive=True,
+        )
 
+        # read validation data from CHRTOUT files
+        validation_data = nhd_io.get_ql_from_wrf_hydro_mf(
+            validation_files,
+            parity_parameters["parity_check_file_index_col"],
+            parity_parameters["parity_check_file_value_col"],
+            # [compare_node],
+        )
+    
+    if "parity_check_file" in parity_parameters.keys():
+        validation_data = pd.read_csv(parity_parameters["parity_check_file"], index_col=0)
+        validation_data.index = validation_data.index.astype(int)
+        validation_data.columns = validation_data.columns.astype("datetime64[ns]")
+        validation_data = validation_data.sort_index(axis="index")
+        
     compare_node = parity_parameters["parity_check_compare_node"]
```

Comment:
Originally, `validation_data` was constructed from a series of CHRTOUT files stored in `parity_parameters["parity_check_input_folder"]`. I added an option to read validation data straight from a .csv file. This happens if `parity_parameters["parity_check_file"]` is provided.

Reply:
@awlostowski-noaa
The code above shows the logic you describe, but it looks like it was lost in a later commit... I'm working on restoring it here for a merge candidate.